packages/gestalt/src/Checkbox.js

```diff
@@ -57,68 +63,100 @@ export default function Checkbox({
-        ref={inputElement}
-        type="checkbox"
-      />
-      <div
-        className={classnames(
-          borderStyle,
-          styles.check,
-          // eslint-disable-next-line no-nested-ternary
-          disabled
-            ? checked || indeterminate
-              ? colors.grayBg
-              : colors.lightGrayBg
-            : checked || indeterminate
-            ? colors.darkGrayBg
-            : colors.whiteBg,
-          {
-            [styles.checkEnabled]: !disabled,
-            [styles.checkFocused]: focused,
-            [styles.checkMd]: size === 'md',
-            [styles.checkSm]: size === 'sm',
-          }
-        )}
+    <Box>
+      <Box
+        alignItems="center"
+        display="flex"
+        justifyContent="start"
+        marginLeft={-1}
+        marginRight={-1}
       >
-        {(checked || indeterminate) && (
-          <Icon
-            accessibilityLabel=""
-            color="white"
-            icon={indeterminate ? 'dash' : 'check'}
-            size={size === 'sm' ? 8 : 12}
-          />
+        <Label htmlFor={id}>
+          <Box paddingX={1} position="relative">
+            <input
+              checked={checked}
+              className={classnames(styles.input, {
+                [styles.inputEnabled]: !disabled,
+                [styles.inputSm]: size === 'sm',
+                [styles.inputMd]: size === 'md',
+              })}
+              disabled={disabled}
+              id={id}
+              name={name}
+              onBlur={() => setFocused(false)}
+              onChange={handleChange}
+              onClick={handleClick}
+              onFocus={() => setFocused(true)}
+              ref={inputElement}
+              type="checkbox"
+            />
+            <div
+              className={classnames(
+                borderStyle,
+                styles.check,
+                // eslint-disable-next-line no-nested-ternary
+                disabled
+                  ? colors.lightGrayBg
+                  : checked || indeterminate
+                  ? colors.darkGrayBg
+                  : colors.whiteBg,
+                {
+                  [styles.checkEnabled]: !disabled,
+                  [styles.checkFocused]: focused,
+                  [styles.checkMd]: size === 'md',
+                  [styles.checkSm]: size === 'sm',
+                }
+              )}
+            >
+              {(checked || indeterminate) && (
+                <Icon
+                  accessibilityLabel=""
+                  color="white"
+                  icon={indeterminate ? 'dash' : 'check'}
+                  size={size === 'sm' ? 8 : 12}
+                />
+              )}
+            </div>
+          </Box>
+        </Label>
+
+        {label && (
+          <Label htmlFor={id}>
+            <Box
+              display="flex"
+              alignItems="center"
+              paddingX={1}
+              height={size === 'sm' ? 16 : 24}
```

Comment:
@AlbertCarreras why did you set a height on this `Box`? It's breaking the vertical center alignment:
![image](https://user-images.githubusercontent.com/127199/77103499-ba239280-69d7-11ea-9c7e-f8a446f69ed6.png)

Can we remove it?

Reply:
ok, let me check it.

Reply:
@AlbertCarreras Following up on this - can we remove the height on this `<Box />`?

Reply:
DONE